pkg/blocks/ttps.go

```diff
@@ -119,228 +118,180 @@ func reduceIndentation(b []byte, n int) []byte {
 	return bytes.Join(lines, []byte("\n"))
 }
 
-// UnmarshalYAML is a custom unmarshalling implementation for the TTP structure.
-// It decodes a YAML Node into a TTP object, handling the decoding and
-// validation of the individual steps within the TTP.
+// Validate ensures that all components of the TTP are valid
+// It checks key fields, then iterates through each step
+// and validates them in turn
 //
 // **Parameters:**
 //
-// node: A pointer to a yaml.Node that represents the TTP structure
-// to be unmarshalled.
+// execCtx: The TTPExecutionContext for the current TTP.
 //
 // **Returns:**
 //
-// error: An error if the decoding process fails or if the TTP structure contains invalid steps.
-func (t *TTP) UnmarshalYAML(node *yaml.Node) error {
-	type TTPTmpl struct {
-		Name        string            `yaml:"name,omitempty"`
-		Description string            `yaml:"description"`
-		Environment map[string]string `yaml:"env,flow,omitempty"`
-		Steps       []yaml.Node       `yaml:"steps,omitempty,flow"`
-		ArgSpecs    []args.Spec       `yaml:"args,omitempty,flow"`
-	}
-
-	var tmpl TTPTmpl
-	if err := node.Decode(&tmpl); err != nil {
-		return err
-	}
```

Comment:
For similar reasons, we no longer need custom unmarshalling of the top-level TTP. Any additional validation required is performed by TTP.Validate()